easybuild/tools/singularity/utilities.py

```diff
@@ -0,0 +1,395 @@
+		pos = def_file.find('.')
+		container_name = def_file[pos+1:]
+
+	#squash image format
+	if image_format == "squashfs":
+		container_name += ".simg"
+		if os.path.exists(container_name):
+			errmsg = "Image already exist at " + os.path.join(singularity_writepath,container_name) 
+			print errmsg
+			EasyBuildError(errmsg)
+			sys.exit(0)
+
+		os.system("sudo singularity build " + container_name + " " + def_file)
+
+	# ext3 image format, creating as writable container 
+	elif image_format == "ext3":
+	    	container_name += ".img"
+
+		if os.path.exists(container_name):
+			errmsg = "Image already exist at " + os.path.join(singularity_writepath,container_name) 
+			print errmsg
+			EasyBuildError(errmsg)
+			sys.exit(0)
+
+		os.system("sudo singularity build --writable " + container_name + " " + def_file)
+
+	# sandbox image format, creates as a directory but acts like a container
+	elif image_format == "sandbox":
+
+		if os.path.exists(container_name):
+			errmsg = "Image already exist at " + os.path.join(singularity_writepath,container_name) 
+			print errmsg
+			EasyBuildError(errmsg)
+			sys.exit(0)
+
+	     	os.system("sudo singularity build --sandbox " + container_name + " " + def_file)
+
+
+    return 
+
+
+
+
+def check_singularity(ordered_ecs,bootstrap_opts,easyconfig_repo,easyblock_repo):
+    """
+    Return build statistics for this build
+    """
+
+    path_to_singularity_cmd = which("singularity")
+    singularity_version = 0
+    if path_to_singularity_cmd:
+	print "Singularity tool found at %s" % path_to_singularity_cmd
+	ret = subprocess.Popen("singularity --version", shell=True,stdin=subprocess.PIPE,stdout=subprocess.PIPE,stderr=subprocess.PIPE)
+	# singularity version format for 2.3.1 and higher is x.y-dist
+	singularity_version = ret.communicate()[0].split("-")[0]
+    else:
+	print "Singularity not found in your system."
+ 	raise EasyBuildError("Singularity not found in your system")
+
+
+    if float(singularity_version) < 2.4:
+    	raise EasyBuildError("Please upgrade singularity instance to version 2.4 or higher")
+
+    else:
+	print "Singularity version is 2.4 or higher ... OK"
+	print "Singularity Version is " + singularity_version
+
+# ---------- uncomment section below when enabling architecture detection and labels in singularity
+
+    #buildsystem_session = session_state()
+    #system_info = buildsystem_session['system_info']
+
+    #ret = subprocess.Popen("""lscpu | grep Model: | cut -f2 -d ":" """,shell=True,stdin=subprocess.PIPE,stdout=subprocess.PIPE,stderr=subprocess.PIPE)
+    #model_num = int(ret.communicate()[0])
+
+    # convert decimal to hex. Output like  0x3e. Take everything after x and convert to uppercase
+    #model_num = hex(model_num).split('x')[-1].upper()
+    #arch_name = architecture_query(model_num)
+
+    generate_singularity_recipe(ordered_ecs, bootstrap_opts,easyconfig_repo,easyblock_repo)
+
+    return 
```

Comment:
Not needed.

Reply:
this method needs to be invoked. As for the arguments if we can find better way to pass these into generate_singularity_recipe I am open to suggestions.

Reply:
I mean the return.